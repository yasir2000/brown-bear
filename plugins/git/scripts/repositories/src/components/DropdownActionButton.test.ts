/*
 * Copyright (c) Enalean 2021 -  Present. All Rights Reserved.
 *
 *  This file is a part of Tuleap.
 *
 * Tuleap is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * Tuleap is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 *
 */

import { createLocalVue, shallowMount } from "@vue/test-utils";
import VueDOMPurifyHTML from "vue-dompurify-html";
import GetTextPlugin from "vue-gettext";
import DropdownActionButton from "./DropdownActionButton.vue";

describe("DropdownActionButton", () => {
    it("displays a dropdown for empty state", () => {
        const localVue = createLocalVue();
        localVue.use(VueDOMPurifyHTML);
        localVue.use(GetTextPlugin, {
            translations: {},
            silent: true,
        });

        const wrapper = shallowMount(DropdownActionButton, {
            localVue,
            propsData: {
                is_empty_state: true,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it("displays a dropdown for repository list", () => {
        const localVue = createLocalVue();
        localVue.use(VueDOMPurifyHTML);
        localVue.use(GetTextPlugin, {
            translations: {},
            silent: true,
        });

        const wrapper = shallowMount(DropdownActionButton, {
            localVue,
            propsData: {
                is_empty_state: false,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });
});
