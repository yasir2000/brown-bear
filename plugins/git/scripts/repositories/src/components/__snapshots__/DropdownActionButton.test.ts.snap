// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownActionButton displays a dropdown for empty state 1`] = `
<div class="tlp-dropdown"><button type="button" class="tlp-button-primary" data-dropdown="trigger">
    <translate-stub tag="span">New repository</translate-stub> <i aria-hidden="true" class="fas fa-caret-down tlp-button-icon-right"></i>
  </button>
  <div role="menu" class="tlp-dropdown-menu" data-dropdown="menu"><button type="button" data-test="create-repository-button" class="tlp-dropdown-menu-item"><i class="fa fa-plus tlp-button-icon"></i>
      <translate-stub tag="span" class="git-add-action-button">Create a repository</translate-stub>
    </button>
    <add-gitlab-repository-action-button-stub></add-gitlab-repository-action-button-stub>
  </div>
</div>
`;

exports[`DropdownActionButton displays a dropdown for repository list 1`] = `
<div class="tlp-dropdown git-repository-list-create-repository-button"><button type="button" class="tlp-button-primary" data-dropdown="trigger">
    <translate-stub tag="span">New repository</translate-stub> <i aria-hidden="true" class="fas fa-caret-down tlp-button-icon-right"></i>
  </button>
  <div role="menu" class="tlp-dropdown-menu" data-dropdown="menu"><button type="button" data-test="create-repository-button" class="tlp-dropdown-menu-item"><i class="fa fa-plus tlp-button-icon"></i>
      <translate-stub tag="span" class="git-add-action-button">Create a repository</translate-stub>
    </button>
    <add-gitlab-repository-action-button-stub></add-gitlab-repository-action-button-stub>
  </div>
</div>
`;
